spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.spark;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import joptsimple.internal.Strings;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Spark plugin Utility class. Contains common code to be used in trainer and predictors.
+ */
+final class SparkUtils {
+
+  private SparkUtils() {
+  }
+
+  /**
+   * Validate the config parameters for the spark sink and spark compute classes.
+   *
+   * @param inputSchema       schema of the received record.
+   * @param featuresToInclude features to be used for training/prediction.
+   * @param featuresToExclude features to be excluded when training/predicting.
+   * @param predictionField   field containing the prediction values.
+   */
+  protected static void validateConfigParameters(Schema inputSchema, @Nullable String featuresToInclude,
```

Comment:
Doesn't make sense to have `protected` modified in util class, since there will be no one inheriting this class. Since this class is package private, non-private methods in this class should also be package private.